packages/theme/src/typography.ts

```diff
@@ -1,26 +1,100 @@
-        fontWeight: fontWeights.regular,
-        fontFamily: fontFamilies.base,
-    },
-    title3: {
-        fontSize: fontSizes.title3,
-        lineHeight: lineHeights.small,
-        fontWeight: fontWeights.regular,
-        fontFamily: fontFamilies.base,
+export const fontWeights = {
+    medium: 400,
+    demi: 600,
+} as const;
+
+export type FontWeight = keyof typeof fontWeights;
+export type FontWeightValue = typeof fontWeights[FontWeight];
+
+// we need unit-less typography base because RN is unit-less, we can easily add units later
+// for web we need string instead of object because styled-components syntax
+export const typographyStylesBase = {
+    largeTitle: {
+        fontSize: 48,
+        lineHeight: 53,
+        fontWeight: fontWeights.medium,
+        letterSpacing: 0.4,
+    },
+    mediumTitle: {
+        fontSize: 34,
+        lineHeight: 37,
+        fontWeight: fontWeights.medium,
+        letterSpacing: -1.4,
+    },
+    smallTitle: {
+        fontSize: 22,
+        lineHeight: 26,
+        fontWeight: fontWeights.medium,
+        letterSpacing: -0.3,
+    },
+    highlight: {
+        fontSize: 16,
+        lineHeight: 24,
+        fontWeight: fontWeights.demi,
+        letterSpacing: -0.4,
+    },
+    body: {
+        fontSize: 16,
+        lineHeight: 24,
+        fontWeight: fontWeights.medium,
+        letterSpacing: -0.4,
+    },
+    callout: {
+        fontSize: 14,
+        lineHeight: 21,
+        fontWeight: fontWeights.demi,
+        letterSpacing: -0.3,
+    },
+    hint: {
+        fontSize: 14,
+        lineHeight: 21,
+        fontWeight: fontWeights.medium,
+        letterSpacing: -0.3,
+    },
+    label: {
+        fontSize: 12,
+        lineHeight: 18,
+        fontWeight: fontWeights.medium,
+        letterSpacing: -0.1,
     },
 } as const;
+
+export type TypographyStyle = keyof typeof typographyStylesBase;
+export type TypographyStyles = Record<TypographyStyle, string>;
+export type NativeTypographyStyleDefinition = {
+    fontSize: number;
+    lineHeight: number;
+    fontWeight: FontWeightValue;
+    letterSpacing: number;
+    fontFamily?: string;
+};
+export type NativeTypographyStyles = Record<TypographyStyle, NativeTypographyStyleDefinition>;
+
+export const prepareTypography = (): TypographyStyles => {
+    const typography: any = {};
+    // eslint-disable-next-line no-restricted-syntax
+    for (const [styleName, value] of Object.entries(typographyStylesBase)) {
+        // for tagged template styles on web we need just string snippet
+        typography[styleName] = `
+            font-size: ${value.fontSize}px;
+            line-height: ${value.lineHeight}px;
+            font-weight: ${value.fontWeight};
+            letter-spacing: ${value.letterSpacing}px;
+            font-family: ${fontFamilies.base}
+        `;
+    }
+    return typography;
+};
+
+export const prepareNativeTypography = (): NativeTypographyStyles => {
+    const typography: any = {};
```

Comment:
why not `NativeTypographyStyles`?

Reply:
Object is empty at init and it's filled one by one in for loop below so it will throw if `NativeTypographyStyles` because that type require all styles to be defined. I could use ` Partial<NativeTypographyStyles>` but I will require even more types below in loop because of implicit any and it will improve type safety very marginally because return type is ensured anyway. So `any` is good compromise here.